var/spack/repos/builtin/packages/py-dask-glm/package.py

```diff
@@ -15,10 +15,33 @@ class PyDaskGlm(PythonPackage):
 
     version('0.2.0', sha256='58b86cebf04fe5b9e58092e1c467e32e60d01e11b71fdc628baaa9fc6d1adee5')
 
+    variant('docs', default=False, description='Build HTML documentation')
+
     depends_on('py-setuptools', type='build')
     depends_on('py-setuptools-scm', type='build')
     depends_on('py-cloudpickle@0.2.2:', type=('build', 'run'))
     depends_on('py-dask+array', type=('build', 'run'))
     depends_on('py-multipledispatch@0.4.9:', type=('build', 'run'))
     depends_on('py-scipy@0.18.1:', type=('build', 'run'))
-    depends_on('py-scikit-learn@0.18:', type=('build', 'run'))
+    depends_on('py-scikit-learn@0.18:', type=('build', 'run'), when='~docs')
+    depends_on('py-scikit-learn@0.18:0.21.999', type='build', when='+docs')
```

Comment:
```suggestion
    depends_on('py-scikit-learn@0.18:0.21.999', type=('build', 'run'), when='+docs')
```
Even if docs are built, sklearn is still needed at run-time.